pkg/registry/service/rest.go

```diff
@@ -157,7 +158,7 @@ func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		}
 	}
 
-	if featuregate.DefaultFeatureGate.ExternalTrafficLocalOnly() && shouldCheckOrAssignHealthCheckNodePort(service) {
+	if shouldCheckOrAssignHealthCheckNodePort(service) {
 		var healthCheckNodePort int
 		var err error
 		if l, ok := service.Annotations[apiservice.AnnotationHealthCheckNodePort]; ok {
```

Comment:
Doesn't `GetServiceHealthCheckNodePort` do this for you? maybe you just return an err and int32 as the distinction between "no node port specified" and "node port specified but couldn't parse".  What do we think should happen if user sets it to 0? currently we clobber it. Maybe we should throw an error instead? 

Leave it to you to do what you think is cleaner, just bringing it up because I observed duplicated code.